packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -261,14 +273,18 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 	}
 
 	public startTransaction(repairStore?: RepairDataStore): void {
-		this.transactions.push(this.editManager.getLocalBranchHead().revision, repairStore);
+		this.transactions.push(
```

Comment:
I think that's where you'll need to clone the forest (when the transaction stack goes from 0 to 1).

Reply:
I ended up doing the cloning within SharedTree so that SharedTreeCore only needs to know that it should have a RepairDataStore that works for the resulting squashed transaction.

Reply:
Right, forest has no place in core, sorry.